### ISSUE TITLE
running_actions_manager_test fails on NixOS

### ISSUE THREAD

**Issue by jaroeichler** [TraceMachina/nativelink#1373]
The tests fail with the following output

``` test entrypoint_injects_properties ... FAILED
test entrypoint_does_invoke_if_set ... FAILED
test entrypoint_sends_timeout_via_side_channel ... ok

failures:

---- unix_executable_file_test stdout ----
thread 'unix_executable_file_test' panicked at nativelink-worker/tests/running_actions_manager_test.rs:2936:35:
index out of bounds: the len is 0 but the index is 0
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- upload_dir_and_symlink_test stdout ----
thread 'upload_dir_and_symlink_test' panicked at nativelink-worker/tests/running_actions_manager_test.rs:1089:37:
index out of bounds: the len is 0 but the index is 0

---- entrypoint_injects_properties stdout ----
thread 'entrypoint_injects_properties' panicked at nativelink-worker/tests/running_actions_manager_test.rs:1725:5:
assertion failed: `(left == right)`: Exit code should be 0

Diff < left / right > :
<127
>0



---- entrypoint_does_invoke_if_set stdout ----
thread 'entrypoint_does_invoke_if_set' panicked at nativelink-worker/tests/running_actions_manager_test.rs:1558:5:
assertion failed: `(left == right)`: Exit code should be 0

Diff < left / right > :
<127
>0




failures:
    entrypoint_does_invoke_if_set
    entrypoint_injects_properties
    unix_executable_file_test
    upload_dir_and_symlink_test

test result: FAILED. 21 passed; 4 failed; 1 ignored; 0 measured; 0 filtered out; finished in 0.29s
```

**Comment by MarcusSorealheis** [comment]
@jaroeichler can you provide more information on your environment? My apologies for getting back to you so late. I've been away.

**Comment by jaroeichler** [comment]
Sure, it's based on #1287. System paths are passed to Bazel with
`build --action_env` and `build --host_action_env` and the link loader is handled by `nix-ld`. Since NixOS uses a non-standard file hierarchy, some runtime dependencies may not be found, potentially causing the test to fail. This should only happen if there's hard coded paths in the executed code.

I'll also look into it myself once I find the time.

**Comment by jaroeichler** [comment]
Fixed by #1427, thanks. 53 out of 53 tests passing on NixOS.
